cmd/zetae2e/local/local.go

```diff
@@ -318,6 +325,52 @@ func localE2ETest(cmd *cobra.Command, _ []string) {
 
 	logger.Print("✅ e2e tests completed in %s", time.Since(testStartTime).String())
 
+	if testMigration {
+		migrationCtx, cancel := context.WithCancel(context.Background())
+		deployerRunner.CtxCancel = cancel
+
+		migrationStartTime := time.Now()
+		logger.Print("🏁 starting tss migration")
+
+		response, err := deployerRunner.CctxClient.LastZetaHeight(
+			migrationCtx,
+			&crosschaintypes.QueryLastZetaHeightRequest{},
+		)
+		require.NoError(deployerRunner, err)
+		err = zetaTxServer.UpdateKeygen(response.Height)
+		require.NoError(deployerRunner, err)
+
+		// Generate new TSS
+		waitKeygenHeight(migrationCtx, deployerRunner.CctxClient, deployerRunner.ObserverClient, logger, 0)
+
+		// migration test is a blocking thread, we cannot run other tests in parallel
+		// The migration test migrates funds to a new TSS and then updates the TSS address on zetacore.
+		// The necessary restarts are done by the zetaclient supervisor
+		fn := migrationTestRoutine(conf, deployerRunner, verbose, e2etests.TestMigrateTssEthName)
+
+		if err := fn(); err != nil {
+			logger.Print("❌ %v", err)
+			logger.Print("❌ tss migration failed")
+			os.Exit(1)
+		}
+
+		logger.Print("✅ migration completed in %s ", time.Since(migrationStartTime).String())
+		logger.Print("🏁 starting post migration tests")
+
+		tests := []string{
+			e2etests.TestBitcoinWithdrawSegWitName,
+			e2etests.TestEtherWithdrawName,
+		}
+		eg.Go(postMigrationTestRoutine(conf, deployerRunner, verbose, tests...))
+
+		if err := eg.Wait(); err != nil {
```

Comment:
Why using a wait group here if only one process? While we actually don't do it above
```
fn := migrationTestRoutine(conf, deployerRunner, verbose, e2etests.TestMigrateTssEthName)

		if err := fn(); err != nil {
```